niworkflows/utils/images.py

```diff
@@ -127,3 +127,83 @@ def dseg_label(in_seg, label, newpath=None):
     new.set_data_dtype(np.uint8)
     new.to_filename(out_file)
     return out_file
+
+
+def resample_by_spacing(in_file, zooms, order=3, clip=True):
```

Comment:
is this function compatible with 4D images? if not, we might want to add a check JIC

Reply:
I think it is, but the test would be great.

Reply:
Okay, it is not compatible with 4D. NiTransforms needs some updates.